plugins/jenkins/src/components/BuildsPage/lib/CITable/CITable.tsx

```diff
@@ -279,17 +281,22 @@ export const CITableView = ({
           <Typography variant="h6">Projects</Typography>
         </Box>
       }
-      columns={generatedColumns}
+      columns={columns && columns.length !== 0 ? columns : generatedColumns}
     />
   );
 };
 
-export const CITable = () => {
+type CITableProps = {
+  columns?: TableColumn[];
```

Comment:
I agree and I also think it is very unlikely that the TableColumn type will ever accept anything other than Project objects in the future. Therefore, I am planning to change the type of the TableColumn array to TableColumn<Project>[]. This will make the code more explicit and easier to maintain.